staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -633,6 +646,11 @@ func (td *typeDiscoverer) discoverStruct(thisNode *typeNode, fldPath *field.Path
 		}
 
 		fields = append(fields, child)
+		return nil
+	})
+
+	if err != nil {
```

Comment:
For readability - this being far away from definition made me go look it up.  What about restructuring this block like:

```
doOneField := func(memb types.Member, name, jsonName string) error {
   // alllllll that code
}
if err := forEachField(thisNode.valueType, doOneField); err != nil {
    return err
}
```

The inner one is shorter but could maybe use the same treatment?

Reply:
Done for both forEachField calls